### ISSUE TITLE
[Bug]: Unable to interact with app after using file picker dialog

### ISSUE THREAD

**Issue by beefcakefu** [electron/electron#35821]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

20.1.4

### What operating system are you using?

Ubuntu

### Operating System Version

Ubuntu 22.04 (Linux UM425IA 5.15.0-48-generic #54-Ubuntu SMP Fri Aug 26 13:26:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux)

### What arch are you using?

x64

### Expected Behavior

After using file picker, focus is expected to return to the calling window and page elements and menu bar should remain responsive.

### Actual Behavior

After using file picker, the entire window becomes unresponsive and would only respond to the close window `X` icon and `Alt+F4` command.

### Testcase Gist URL

https://gist.github.com/scottnonnenberg-signal/fc5174e6015c61fdae49ffe767d441bc

### Additional Information

A bug report was initially filed against https://github.com/signalapp/Signal-Desktop/issues/6115 and some preliminary troubleshooting steps were covered.

Output of `wmctrl`:

    $ wmctrl -m
    Name: GNOME Shell
    Class: N/A
    PID: N/A
    Window manager's "showing the desktop" mode: OFF

Issue was not resolved in Signal Desktop app by using `--disable-gpu` flag.

Issue persists across both `X11` and `Wayland` DEs.

The hardware on which this issue occurs is AMD Ryzen 7 4700U with Radeon graphics, detailed specs [here](https://www.asus.com/Laptops/For-Home/Zenbook/Zenbook-14-UM425/techspec/). Ubuntu 22.04 is installed on bare metal as the only operating system.

In the Fiddle, dev tools also becomes unresponsive when the bug occurs, except when undocked into separate window.

Using the dev tools cursor icon, where different page elements would be highlighted when hovering over them normally, they stop highlighting when the bug occurs.

I have confirmed that the bug exists in Electron versions `19.0.6` and `20.1.4`.

When the bug occurs ping/pong activity in dev tools console are unaffected.

Packaged versions are unaffected when opened in my browser (Firefox 102.3.0esr 64-bit).

![Screenshot from 2022-09-24 14-53-40](https://user-images.githubusercontent.com/103491197/192351359-f832ffc2-1e98-4308-86cf-6832cbebd6ff.png)

**Comment by beefcakefu** [comment]
I realize the Fiddle remains responsive if an image or file is selected by dragging it over the `Choose File` button, instead of using the file picker.

**Comment by eager-signal** [comment]
I also encountered this bug recently. It reproduces, both when choose a file, and when I cancel the open file dialog.

Using Electron Fiddle’s bisect, the issue was introduced by these commits: https://github.com/electron/electron/compare/v16.0.0-nightly.20210913...v16.0.0-nightly.20210914

During the full reproduction steps, including after the dialog closes and the fiddle is unresponsive, this is the output of `/var/log/syslog`:
```
Jan  6 11:13:00 systemd[1]: Started Hostname Service.
Jan  6 11:13:46 systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Jan  6 11:14:21 systemd[1998]: vte-spawn-cfb2859f-8964-4ce9-a466-de6322d4cbe4.scope: Consumed 41.215s CPU time.
Jan  6 11:14:21 systemd[1998]: gnome-terminal-server.service: Consumed 2.634s CPU time.
Jan  6 11:14:35 systemd[1998]: Started Application launched by gnome-shell.
Jan  6 11:14:35 dbus-daemon[2021]: [session uid=1000 pid=2021] Activating via systemd: service name='org.gnome.Terminal' unit='gnome-terminal-server.service' requested by ':1.420' (uid=1000 pid=12724 comm="/usr/bin/gnome-terminal.real " label="unconfined")
Jan  6 11:14:35 systemd[1998]: Starting GNOME Terminal Server...
Jan  6 11:14:35 dbus-daemon[2021]: [session uid=1000 pid=2021] Successfully activated service 'org.gnome.Terminal'
Jan  6 11:14:35 systemd[1998]: Started GNOME Terminal Server.
Jan  6 11:14:35 systemd[1998]: Started VTE child process 12747 launched by gnome-terminal-server process 12729.
```

As for my system, this is also on 22.04, Linux 5.15.0-57-generic, x86_64.
```
$ wmctrl -m
Name: GNOME Shell
Class: N/A
PID: N/A
Window manager's "showing the desktop" mode: OFF
```

**Comment by wrdhub** [comment]
I have also recently run into this issue.

I am on Ubuntu 22.04, Linux 5.15.0-56-generic x86_64. I have also tried X11 and Wayland with the same result.

The issue happens consistently on all desktop clients that I use that use electron (Tutanota, Signal, and Wire).

beefcakefu's workaround of dragging and dropping files works for me too, so I guess I will be doing that for now.

**Comment by TheSecurityDev** [comment]
I've noticed a similar issue on Discord and VS Code in Windows 10, where opening the file picker, then canceling without selecting anything and opening again causes the app to hang.

**Comment by lc-guy** [comment]
Hitting this bug too, over multiple Electron apps. Up-to-date Ubuntu 22.04

**Comment by beefcakefu** [comment]
This was resolved for me in Signal Desktop app (6.24.0) for Linux.

**Comment by TheSecurityDev** [comment]
Still broken for me on Signal, Windows 10.
